integrationTests/testConsensusNode.go

```diff
@@ -0,0 +1,434 @@
+type TestConsensusNode struct {
+	Node             *node.Node
+	Messenger        p2p.Messenger
+	NodesCoordinator nodesCoordinator.NodesCoordinator
+	ShardCoordinator sharding.Coordinator
+	ChainHandler     data.ChainHandler
+	BlockProcessor   *mock.BlockProcessorMock
+	ResolverFinder   dataRetriever.ResolversFinder
+	AccountsDB       *state.AccountsDB
+	NodeKeys         TestKeyPair
+}
+
+// NewTestConsensusNode returns a new TestConsensusNode
+func NewTestConsensusNode(
+	consensusSize int,
+	roundTime uint64,
+	consensusType string,
+	nodeKeys TestKeyPair,
+	eligibleMap map[uint32][]nodesCoordinator.Validator,
+	waitingMap map[uint32][]nodesCoordinator.Validator,
+	keyGen crypto.KeyGenerator,
+) *TestConsensusNode {
+
+	shardCoordinator, _ := sharding.NewMultiShardCoordinator(maxShards, nodeShardId)
+
+	tcn := &TestConsensusNode{
+		NodeKeys:         nodeKeys,
+		ShardCoordinator: shardCoordinator,
+	}
+	tcn.initNode(consensusSize, roundTime, consensusType, eligibleMap, waitingMap, keyGen)
+
+	return tcn
+}
+
+// CreateNodesWithTestConsensusNode returns a map with nodes per shard each using TestConsensusNode
+func CreateNodesWithTestConsensusNode(
+	numMetaNodes int,
+	nodesPerShard int,
+	consensusSize int,
+	roundTime uint64,
+	consensusType string,
+) map[uint32][]*TestConsensusNode {
+
+	nodes := make(map[uint32][]*TestConsensusNode, nodesPerShard)
+	cp := CreateCryptoParams(nodesPerShard, numMetaNodes, maxShards)
+	keysMap := PubKeysMapFromKeysMap(cp.Keys)
+	validatorsMap := GenValidatorsFromPubKeys(keysMap, maxShards)
+	eligibleMap, _ := nodesCoordinator.NodesInfoToValidators(validatorsMap)
+	waitingMap := make(map[uint32][]nodesCoordinator.Validator)
+	connectableNodes := make([]Connectable, 0)
+
+	for _, keysPair := range cp.Keys[0] {
+		tcn := NewTestConsensusNode(
+			consensusSize,
+			roundTime,
+			consensusType,
+			*keysPair,
+			eligibleMap,
+			waitingMap,
+			cp.KeyGen)
+		nodes[nodeShardId] = append(nodes[nodeShardId], tcn)
+		connectableNodes = append(connectableNodes, tcn)
+	}
+
+	ConnectNodes(connectableNodes)
+
+	return nodes
+}
+
+func (tcn *TestConsensusNode) initNode(
+	consensusSize int,
+	roundTime uint64,
+	consensusType string,
+	eligibleMap map[uint32][]nodesCoordinator.Validator,
+	waitingMap map[uint32][]nodesCoordinator.Validator,
+	keyGen crypto.KeyGenerator,
+) {
+
+	testHasher := createHasher(consensusType)
+	epochStartRegistrationHandler := notifier.NewEpochStartSubscriptionHandler()
+	consensusCache, _ := lrucache.NewCache(10000)
+	pkBytes, _ := tcn.NodeKeys.Pk.ToByteArray()
+
+	tcn.initNodesCoordinator(consensusSize, testHasher, epochStartRegistrationHandler, eligibleMap, waitingMap, pkBytes, consensusCache)
+	tcn.Messenger = CreateMessengerWithNoDiscovery()
+	tcn.initBlockChain(testHasher)
+	tcn.initBlockProcessor()
+
+	startTime := time.Now().Unix()
+
+	singlesigner := &ed25519SingleSig.Ed25519Signer{}
+	singleBlsSigner := &mclsinglesig.BlsSingleSigner{}
```

Comment:
wondering why we don't have these 2 signers exported as constants in `integrationTests` package

Reply:
extracted and refactored multiple tests